metadata/src/main/java/org/apache/kafka/controller/FeatureControlManager.java

```diff
@@ -264,11 +265,11 @@ FinalizedControllerFeatures finalizedFeatures(long epoch) {
     }
 
     public void replay(FeatureLevelRecord record) {
-        if (!canSupportVersion(record.name(), record.featureLevel())) {
```

Comment:
This made me wonder about bootstrapping and if we could load in an unknown version. However, looking at BootstrapMetadata#load, we will parse the feature level as a MetadataVersion before returning so we would throw an exception as we read the bootstrap records into memory. This could happen if someone generates a bootstrap snapshot at version N and then tries to load it into a cluster at N-1

Reply:
Yeah.

It's fine to do a sanity check based on local supported versions. Pulling in the quorum is where it is incorrect (which might be an argument for pulling `localSupportedFeatures` out of `QuorumFeatures`, but one step at a time.)